cli/mender-artifact/dump.go

```diff
@@ -184,18 +184,22 @@ func printCmdline(ar *areader.Reader, args []string) {
 	fmt.Printf("write module-image")
 
 	artProvs := ar.GetArtifactProvides()
-	fmt.Printf(" --artifact-name %s", artProvs.ArtifactName)
-	if len(artProvs.ArtifactGroup) > 0 {
-		fmt.Printf(" --provides-group %s", artProvs.ArtifactGroup)
+	if artProvs != nil {
+		fmt.Printf(" --artifact-name %s", artProvs.ArtifactName)
```

Comment:
I think there is a subtle error here. `ArtifactName` was not stored in provides in v2, it was stored in its own field. You might want to fall back to it.

The same goes for `device-type` below.

V2 is not important to keep around, so it's also ok not to bother with it, but if so, an error is a better outcome here, since the dump won't be correct.

Reply:
@kacf Thanks; good catch!

I decided for the former, correctly print `--artifact-name` and `--device-type` on v2 Artifacts